### PR TITLE
Actor based TPP sdk change

### PR DESCRIPTION
**Pull request by vanika-ap**

### CODE REVIEW

File: Controllers/ApplicationController.cs [hunk 1]
```diff
@@ -148,19 +95,67 @@ public Task<string> Redeem()
...
+               // check CSRF token and retrieve state and token ID from callback parameters
+               .FlatMap(mem => tokenClient.ParseTokenRequestCallbackUrl(Request.Url.AbsoluteUri, csrfToken.Value)
+                   // get the token and check its validity
+                   .FlatMap(callback => mem.GetToken(callback.TokenId))
+                   // redeem the token at the server to move the funds
+                   .FlatMap(mem.RedeemToken)
+                   .Map(transfer => "Success! Redeemed transfer " + transfer.Id));
+        }
+
+        private static Task<string> InitializeTokenRequestUrl(
```
Comment (gianluca-token) [inline Controllers/ApplicationController.cs]: I think this method does way too much. It stores the token request, sets a browser cookie and generates the token request url, so it's really hard to come up with a name that encompasses that. Maybe you can just use a method that converts the queryData into a token request?

File: Controllers/ApplicationController.cs [hunk 2]
```diff
@@ -41,100 +43,45 @@ public ActionResult Index()
...
 
-            // generate a reference ID for the token
-            var refId = Util.Nonce();
-
-            var cookie = new HttpCookie("csrf_token") {Value = csrfToken};
-            // set CSRF token in browser cookie
-            Response.Cookies.Add(cookie);
+            formData.GetType().GetProperties()
+                .ToList()
+                .ForEach(pi => queryData.Add(pi.Name, pi.GetValue(formData, null)?.ToString()));
```
Comment (gianluca-token) [inline Controllers/ApplicationController.cs]: why pi?

File: Controllers/ApplicationController.cs (hunk 1 above)
Comment (mahendra-aps) [reply Controllers/ApplicationController.cs]: We followed same format as it is used in java merchant sample. Do we still need to change this

Comment (gianluca-token) [reply Controllers/ApplicationController.cs]: Oh interesting... no you're fine then.

**Pull request merged**